### ISSUE TITLE
Deployment does not work with kubectl 1.23.0

### ISSUE THREAD

**Issue by toddq** [loft-sh/devspace#1813]
**What happened?**  
Using devspace deployment with kubectl results in 

```
[fatal]  error deploying: error deploying server: error: --dry-run flag without a value was specified. A value must be set: "none", "server", or "client".
```

`kubectl --dry-run` has been deprecated awhile and was removed in kubectl 1.23.0 - 
https://github.com/kubernetes/kubernetes/pull/105327

**What did you expect to happen instead?**  

Deployment to at least be attempted.

**How can we reproduce the bug?** (as minimally and precisely as possible)  

- Install kubectl 1.23.0
- Configure devspace with a kubectl deployment manifest
- Run `devspace dev`

**Local Environment:**  
- DevSpace Version: 5.17.0
- Operating System:  mac
- Deployment method: kubectl apply

**Kubernetes Cluster:**  
- Cloud Provider: google | aws | azure | DevSpace Cloud | DevSpace Cloud Enterprise | other
- Kubernetes Version: [use `kubectl version`]

**Anything else we need to know?**  

Workaround - I was able to roll back my `kubernetes-cli` version to 1.22.4 (not easy with Homebrew) and things work as expected now.

/kind bug

**Comment by pratikjagrut** [comment]
@toddq Thanks for creating this issue. Yes, kubectl has deprecated the --dry-run. We'll fix that with the next release

**Comment by pratikjagrut** [comment]
Hi @toddq we've released the fix for this issue in v5.18.0. Can you please check and let us know if we can close this issue?

**Comment by toddq** [comment]
Works great, thank you!  Close away.
